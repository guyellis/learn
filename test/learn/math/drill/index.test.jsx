const React = require('react');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const Drill = require('../../../../src/learn/math/drill');

const muiTheme = getMuiTheme(lightBaseTheme);

jest.mock('material-ui/internal/EnhancedSwitch');

describe('Drill', () => {
  beforeEach(() => {
    localStorage.setItem = jest.fn();
    localStorage.getItem = jest.fn();
  });

  afterEach(() => {
    localStorage.setItem.mockClear();
    localStorage.getItem.mockClear();
  });

  test('should render the Drill component', () => {
    const savedOptions = {
      largeKeyboard: true,
      levelIndex: 0, // A
      minutes: '1',
      onscreenKeyboard: true,
      opIndexes: [0], // A
      totalProblems: '20',
      userName: 'my name',
    };
    localStorage.getItem.mockReturnValue(JSON.stringify(savedOptions));

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Drill />
      </MuiThemeProvider>);

    const app = component.toJSON();
    expect(app).toMatchSnapshot();
  });
});
