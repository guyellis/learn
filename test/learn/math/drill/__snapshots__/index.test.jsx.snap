// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drill should render the Drill component 1`] = `
<div>
  <div
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <h2>
      Level
    </h2>
    <div>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={true}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          A
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          B
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          C
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          D
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          E
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          F
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          G
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          H
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          I
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          J
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          K
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          L
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          M
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          N
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          O
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          P
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          Q
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          R
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          S
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          T
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          U
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          V
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          W
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          X
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          Y
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          Z
        </span>
      </button>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <h2>
      Operation
    </h2>
    <div>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={true}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          +
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          -
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          x
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiFab-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        secondary={false}
        style={
          Object {
            "fontSize": "2em",
            "margin": "5px",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          ÷
        </span>
      </button>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <h3>
      Time
    </h3>
    <div
      className="MuiFormControl-root MuiTextField-root"
      style={
        Object {
          "marginLeft": 20,
          "width": 100,
        }
      }
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        htmlFor="time-minutes"
      >
        Time
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="time-minutes-helper-text"
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="time-minutes"
          name="minutes"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
          value="1"
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-filled"
        id="time-minutes-helper-text"
      >
        1
      </p>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <h3>
      Total Questions (you only get badges for 10 or more correct questions)
    </h3>
    <div
      className="MuiFormControl-root MuiTextField-root"
      style={
        Object {
          "marginLeft": 40,
          "width": 150,
        }
      }
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        htmlFor="total-problems"
      >
        Total Questions
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="total-problems-helper-text"
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="total-problems"
          name="totalProblems"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="number"
          value="20"
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-filled"
        id="total-problems-helper-text"
      >
        20
      </p>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <h3>
      Keyboard
    </h3>
    <div
      className="MuiFormGroup-root MuiFormGroup-row"
    >
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          className="MuiSwitch-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-96 MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-checked-97 Mui-checked"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <span
                className="MuiSwitch-thumb"
              />
              <input
                checked={true}
                className="PrivateSwitchBase-input-99 MuiSwitch-input"
                name="onscreenKeyboard"
                onChange={[Function]}
                type="checkbox"
              />
            </span>
          </span>
          <span
            className="MuiSwitch-track"
          />
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Use onscreen keyboard
        </span>
      </label>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <h3>
      Large Keyboard
    </h3>
    <div
      className="MuiFormGroup-root MuiFormGroup-row"
    >
      <label
        className="MuiFormControlLabel-root"
      >
        <span
          className="MuiSwitch-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-96 MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-checked-97 Mui-checked"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <span
                className="MuiSwitch-thumb"
              />
              <input
                checked={true}
                className="PrivateSwitchBase-input-99 MuiSwitch-input"
                name="largeKeyboard"
                onChange={[Function]}
                type="checkbox"
              />
            </span>
          </span>
          <span
            className="MuiSwitch-track"
          />
        </span>
        <span
          className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          Large Keyboard
        </span>
      </label>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <h3>
      Your Name (Optional)
    </h3>
    <div
      className="MuiFormControl-root MuiTextField-root"
      style={
        Object {
          "marginLeft": 40,
          "width": 250,
        }
      }
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        htmlFor="user-name"
      >
        Your Name
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="user-name-helper-text"
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="user-name"
          name="userName"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value="my name"
        />
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-filled"
        id="user-name-helper-text"
      >
        Jemima Puddle-Duck
      </p>
    </div>
  </div>
  <div
    style={
      Object {
        "marginTop": "30px",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Start
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "marginTop": "30px",
      }
    }
  />
</div>
`;
